src/com/oltpbenchmark/DBWorkload.java

```diff
@@ -866,27 +868,25 @@ private static Results runWorkload(List<BenchmarkModule> benchList, boolean verb
 
         }
         Results r = ThreadBench.runRateLimitedBenchmark(workers, workConfs, intervalMonitor);
-        long end = System.nanoTime();
 
         long numNewOrderTransactions = 0;
         for (Worker<?> w : workers) {
             for (LatencyRecord.Sample sample : w.getLatencyRecords()) {
-                if (sample.tranType == newOrderTxnId) {
+                if (sample.tranType == newOrderTxnId && sample.startNs + 1000L * sample.latencyUs <= end) {
                     ++numNewOrderTransactions;
                 }
             }
         }
         LOG.info(SINGLE_LINE);
         LOG.info("Rate limited reqs/s: " + r);
 
-        long time_seconds = (end - start) / 1000 / 1000 / 1000;
-        long tpmc = numNewOrderTransactions * 60 * 1000 * 1000 * 1000 / (end - start);
+        long tpmc = numNewOrderTransactions * 60 / time;
         double efficiency = 1.0 * tpmc * 100 / numWarehouses / 12.86;
 
         LOG.info("Num New Order transactions : " + numNewOrderTransactions +
-                 " time seconds: " + time_seconds +
+                 " time seconds: " + time +
```

Comment:
Let's log these on separate lines for better readability.

Reply:
Done.